src/components/response/ResponseBlock.tsx

```diff
@@ -150,14 +103,9 @@ export default function ResponseBlock({
         )}
         {showNextBtn && (
           <NextButton
-            disabled={
-              hasCorrectAnswer
-                ? !checkClicked
-                : !areResponsesValid
-            }
-            process={() => {
+            disabled={hasCorrectAnswer && !checkClicked}
+            onClick={() => {
```

Comment:
There is an onclick here and in the button itself, is that an issue? Is the order that they execute in fixed? Does that matter?

Reply:
I haven't tested it, but order might be important here.
`onClick` might be dependent on the a ui component. And if it is called after the next screen has been loaded (previous screen is unmounted), we might get an error.